pacai/ui/topGraphics.py

```diff
@@ -0,0 +1,600 @@
+                        outlineColor = color, fillColor = color, width = 1)
+                    imageRow.append(dot)
+                else:
+                    imageRow.append(None)
+        return foodImages
+
+    def drawCapsules(self, capsules):
+        capsuleImages = {}
+        for capsule in capsules:
+            (screen_x, screen_y) = self.to_screen(capsule)
+            dot = graphicsUtils.circle(
+                (screen_x, screen_y), graphicsConstants.CAPSULE_SIZE * self.gridSize,
+                outlineColor = graphicsConstants.CAPSULE_COLOR,
+                fillColor = graphicsConstants.CAPSULE_COLOR, width = 1)
+            capsuleImages[capsule] = dot
+        return capsuleImages
+
+    def removeFood(self, cell, foodImages):
+        x, y = cell
+        graphicsUtils.remove_from_screen(foodImages[x][y])
+
+    def removeCapsule(self, cell, capsuleImages):
+        x, y = cell
+        graphicsUtils.remove_from_screen(capsuleImages[(x, y)])
+
+    def drawExpandedCells(self, cells):
+        """
+        Draws an overlay of expanded grid positions for search agents
+        """
+        n = float(len(cells))
+        baseColor = [1.0, 0.0, 0.0]
+        self.clearExpandedCells()
+        self.expandedCells = []
+        for k, cell in enumerate(cells):
+            screenPos = self.to_screen(cell)
+            cellColor = graphicsUtils.formatColor(*[(n - k) * c * .5 / n + .25 for c in baseColor])
+            block = graphicsUtils.square(screenPos, 0.5 * self.gridSize,
+                    color = cellColor, filled = 1, behind = 2)
+            self.expandedCells.append(block)
+            if self.frameTime < 0:
+                graphicsUtils.refresh()
+
+    def clearExpandedCells(self):
+        if 'expandedCells' in dir(self) and len(self.expandedCells) > 0:
+            for cell in self.expandedCells:
+                graphicsUtils.remove_from_screen(cell)
+
+    def updateDistributions(self, distributions):
+        # Draws an agent's belief distributions
+        if (self.distributionImages is None):
+            self.drawDistributions(self.previousState)
+
+        for x in range(len(self.distributionImages)):
+            for y in range(len(self.distributionImages[0])):
+                image = self.distributionImages[x][y]
+                weights = [dist[(x, y)] for dist in distributions]
+
+                if sum(weights) != 0:
+                    pass
+                # Fog of war
+                color = [0.0, 0.0, 0.0]
+                colors = graphicsConstants.GHOST_VEC_COLORS[1:]  # With Pacman
+                if self.capture:
+                    colors = graphicsConstants.GHOST_VEC_COLORS
+
+                for weight, gcolor in zip(weights, colors):
+                    color = [min(1.0, c + 0.95 * g * weight ** .3) for c, g in zip(color, gcolor)]
+                graphicsUtils.changeColor(image, graphicsUtils.formatColor(*color))
+        graphicsUtils.refresh()
+
+def add(x, y):
```

Comment:
Is add used anywhere else besides in this class? If not add it into the class as a function.

Reply:
Made add() an class function for AbstractPacmanGraphics.